src/drivers/dw/dma.c

```diff
@@ -367,12 +368,12 @@ static void dw_dma_channel_put_unlocked(struct dma *dma, int channel)
 		return;
 	}
 
-	if (!chan->timer_delay) {
-		/* mask block, transfer and error interrupts for channel */
-		dw_write(dma, DW_MASK_TFR, INT_MASK(channel));
-		dw_write(dma, DW_MASK_BLOCK, INT_MASK(channel));
-		dw_write(dma, DW_MASK_ERR, INT_MASK(channel));
-	}
+#ifndef CONFIG_TIMER_SCHEDULING
+	/* mask block, transfer and error interrupts for channel */
+	dw_write(dma, DW_MASK_TFR, INT_MASK(channel));
+	dw_write(dma, DW_MASK_BLOCK, INT_MASK(channel));
+	dw_write(dma, DW_MASK_ERR, INT_MASK(channel));
+#endif
```

Comment:
We don't want to push scheduling detail into dw-dma at build time. It would be far better to pass whether we need to use the DMA IRQ at channel get() time. Looking at usage I can see
```
timer delay = 1

chan_put()
{
	dont mask DMA IRQs
	stop work
}


chan_start()
{
	dont clear DMA IRQs
	schedule work;
}

dma_stop()
{
	cancel_work()
	dont disable IRQs
}

dma_config()
{
	dont unmask IRQs // not aligned with put(), should be in get() ?
	init work
}
```
So easy enough to add a per channel "block_complete_irq" flag to differentiate.